diff --git a/src/components/cuda/tests/Makefile b/src/components/cuda/tests/Makefile
index 57da3d948..9486ed62b 100644
--- a/src/components/cuda/tests/Makefile
+++ b/src/components/cuda/tests/Makefile
@@ -22,9 +22,9 @@ endif
 
 PAPI_FLAG = -DPAPI    # Comment this line for tests to run without PAPI profiling
 NVCFLAGS += -g -ccbin='$(CC)' $(PAPI_FLAG)
-CFLAGS += -g $(PAPI_FLAG)
+CFLAGS += -g $(PAPI_FLAG) -pthread
 INCLUDE += -I$(PAPI_CUDA_ROOT)/include
-CUDALIBS = -L$(PAPI_CUDA_ROOT)/lib64 -lcudart -lcuda
+CUDALIBS = -L$(PAPI_CUDA_ROOT)/lib64 -lcudart -lcuda -lstdc++
 PAPILIB += -L../../../libpfm4/lib -lpfm
 
 cuda_tests: $(TESTS) $(TESTS_NOCTX)
